### PR TITLE
fix(react): fix deep link for telegram's webviews

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description
Fixed Telegram's deep linking logic in order to return to the bot conversation when a webview is closed and added temporal logic for Apple Business Chat and Twitter DMs.

### CODE REVIEW

File: packages/botonic-core/src/models/legacy-types.ts [hunk 1]
```diff
@@ -236,6 +236,7 @@ export interface Params {
 
 export const Providers = Object.freeze({
   Messaging: {
+    APPLE_BUSINESS_CHAT: 'apple_business_chat',
```
Comment (ericmarcos) [inline packages/botonic-core/src/models/legacy-types.ts:239]: In our backend we define it just as "apple", so please change it to:
```
APPLE: 'apple'
```

**Review by ericmarcos — APPROVED** [review]

**Comment by ericmarcos** [comment]
Congrats on your first PR after so long! 👏👏👏

**Review by JianChen-IT — APPROVED** [review]
<3

File: packages/botonic-react/src/webview.jsx [hunk 2]
```diff
@@ -40,8 +40,20 @@ class App extends React.Component {
...
+    if (provider === PROVIDER.TELEGRAM) {
+      location.href = 'https://t.me/' + impId
+    }
+    if (provider === PROVIDER.APPLE) {
+      location.href = 'https://bcrw.apple.com/urn:biz:' + impId
+    }
+    if (provider === PROVIDER.TWITTER) {
+      location.href =
+        'https://twitter.com/messages/compose?recipient_id=' + impId
     } else {
```
[the remark this answers is not in the document]
Comment (ericmarcos) [reply packages/botonic-react/src/webview.jsx:57]: That's right @gbarba !
@vanbasten17 can you please update the final "else" to:

```
    if (provider === PROVIDER.FACEBOOK) {
      try {
        window.MessengerExtensions.requestCloseBrowser(
          () => undefined,
          err => console.log(err)
        )
      } catch (e) {}
    if (provider === PROVIDER.WEBCHAT) {
      try {
        await parent.postMessage('botonicCloseWebview', '*')
      } catch (e) {}
    }
```

**Pull request merged**